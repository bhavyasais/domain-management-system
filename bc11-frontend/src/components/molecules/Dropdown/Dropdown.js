import { FormControl, MenuItem, Select, withStyles } from '@material-ui/core';
import React from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';
const styles = (theme) => ({
  formControl: {
    margin: theme.spacing(1),
    minWidth: 150,
  },
  selectEmpty: {
    marginTop: theme.spacing(2),
  },
  dropdown: {
    height: '38px',
  },
});

const Dropdown = ({ classes, onChange, selectedItem, list, containerStyle, dropDownStyle }) => {
  const handleChange = (event) => {
    onChange?.(event?.target?.value);
  };
  return (
    <div className={containerStyle}>
      <FormControl variant="outlined" className={clsx(classes.formControl, dropDownStyle)}>
        <Select
          value={selectedItem?.value}
          onChange={handleChange}
          labelId="simple-select-label"
          id="simple-select"
          displayEmpty
          className={classes.dropdown}
        >
          <MenuItem value="">None</MenuItem>
          {list.map((item) => (
            <MenuItem value={item?.value} key>
              {item?.name}
            </MenuItem>
          ))}
        </Select>
      </FormControl>
    </div>
  );
};

Dropdown.propTypes = {
  onChange: PropTypes.func,
  selectedItem: PropTypes.shape({
    value: PropTypes.string,
    name: PropTypes.string,
  }),
  list: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string,
      name: PropTypes.string,
    }),
  ),
  containerStyle: PropTypes.shape,
};
Dropdown.defaultProps = {
  onChange: null,
  selectedItem: {
    value: '',
    name: '',
  },
  list: [],
  containerStyle: null,
};

export default withStyles(styles)(Dropdown);
