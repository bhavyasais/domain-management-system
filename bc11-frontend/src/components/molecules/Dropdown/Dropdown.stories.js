import React, { useState } from 'react';
import Dropdown from './Dropdown';

export default {
  title: 'Molecules/DropDown',
};

export const DropDown = () => {
  const [state, setState] = useState({ name: '', value: '' });
  const list = [
    { name: 'zemosolabs', value: 'zemosolabs' },
    { name: 'stanford', value: 'zemostanfordsolabs' },
  ];
  const handleOnChange = (event) => {
    setState({ name: event, value: event });
  };
  return <Dropdown onChange={handleOnChange} selectedItem={state} list={list} />;
};
