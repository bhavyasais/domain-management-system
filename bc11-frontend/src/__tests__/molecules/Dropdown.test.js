import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import Dropdown from '../../components/molecules/Dropdown/Dropdown';
import { Select } from '@material-ui/core';

describe('<SideNavigationBar/>', () => {
  let mounted;
  let onClick;
  const list = [
    { name: 'zemosolabs', value: 'zemosolabs' },
    { name: 'stanford', value: 'zemostanfordsolabs' },
  ];
  beforeEach(() => {
    onClick = jest.fn();
    act(() => {
      mounted = mount(
        <Dropdown list={list} onChange={onClick} selectedItem={{ name: '', value: '' }} />,
      );
    });
  });

  it('Renders correctly', () => {
    let wrapper;
    wrapper = renderer.create(
      <Dropdown list={list} onChange={onClick} selectedItem={{ name: '', value: '' }} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('onClick works on clicking active icon', () => {
    mounted.find(Select).at(0).props().onChange();
    expect(onClick).toHaveBeenCalled();
  });
});
