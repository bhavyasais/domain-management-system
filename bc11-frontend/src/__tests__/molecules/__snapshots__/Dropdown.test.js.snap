// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNavigationBar/> Renders correctly 1`] = `
<div
  className={null}
>
  <div
    className="MuiFormControl-root Dropdown-formControl-1"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root Dropdown-dropdown-3 MuiInputBase-formControl"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="simple-select-label simple-select"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
        id="simple-select"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        None
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput"
        onAnimationStart={[Function]}
        onChange={[Function]}
        required={false}
        tabIndex={-1}
        value=""
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-5"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
